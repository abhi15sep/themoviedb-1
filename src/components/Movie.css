.Movie-card {
    background-color: #fff;
    box-shadow: 0 2px 8px rgba(0,0,0,0.1);
    margin-bottom: 2em;
    min-height: 278px;
}

.Movie-body {
    text-align: left;
    padding: 20px;
    position: relative;
    min-height: 278px;
}

/* Medium devices (desktops, 992px and up) */
@media (min-width: 992px) {
    .Movie-body {
        padding: 10px 20px;
    }
 }

.Movie-title {
    margin-top: 0;
    margin-bottom: 0;
    font-size: 1.2em;
    color: #333;
}

.Movie-overview {
    padding-top: 1em;
}

.Movie-details {
    position: absolute;
    bottom: 0;
    left: 0;
    border-top: solid #eee 1px;
    padding: 1em 1em 1em 1.2em;
    width: 100%;
    display: inline-block;
    color: #666;
}

.Movie-details-link {
    color: #666;
}

.Movie-details:hover {
    text-decoration: none;
}
