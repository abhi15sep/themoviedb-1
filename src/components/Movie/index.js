import React from 'react';
import { Link } from 'react-router-dom';
import {
  Row,
  Col,
  Badge,
} from 'react-bootstrap';
import { FaInfoCircle } from 'react-icons/fa';
import Recommendation from '../Recommendation';
import Image from '../Image';
import Types from '../../utils/types';
import CONSTANTS from '../../utils/constants';
import dateNormalized from '../../utils/dateNormalized';
import getLabelInfo from '../../utils/getLabelInfo';
import './styles.css';

const MAX_OVERVIEW_WORDS = 35;
const { SIZE } = CONSTANTS.IMAGE;

const Movie = ({ data }) => {
  const {
    id,
    title,
    vote_average: voteAverage,
    overview,
    poster_path: posterPath,
    release_date: releaseDate,
  } = data;

  const overviewSmaller = overview
    .split(' ')
    .splice(0, MAX_OVERVIEW_WORDS)
    .join(' ');

  const releaseDateNormalized = dateNormalized(releaseDate);

  const voteAverageNormalized = Math.floor(voteAverage);

  const labelInfo = getLabelInfo(voteAverageNormalized);

  return (
    <Col sm={12} lg={6}>
      <div className="Movie-card">
        <Row>
          <Col sm={12} md={4}>
            <Link to={`/details/${id}`}>
              <Image size={SIZE.SMALL} path={posterPath} title={title} />
            </Link>
          </Col>

          <Col sm={12} md={8}>
            <div className="Movie-body">

              <Row>
                <Col xs={1}>
                  <Badge
                    variant={labelInfo}
                  >
                    {voteAverage}
                  </Badge>
                </Col>
                <Col xs={10}>
                  <Link to={`/details/${id}`}>
                    <span className="Movie-title">{title}</span>
                  </Link>
                  <br />
                  <small>
                    {releaseDateNormalized}
                  </small>
                </Col>
              </Row>

              <p className="Movie-overview">
                {overviewSmaller}
                ...
              </p>

              <div className="Movie-details">
                <Link to={`/details/${id}`} className="Movie-details-link">
                  <FaInfoCircle />
                  &nbsp;
                  <small>
                    More info
                  </small>
                </Link>

                <Recommendation vote={voteAverageNormalized} />
              </div>
            </div>
          </Col>
        </Row>
      </div>
    </Col>
  );
};

Movie.propTypes = {
  data: Types.movie.isRequired,
};

export default Movie;
