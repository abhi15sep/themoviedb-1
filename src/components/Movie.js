import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { Row, Col, Media } from 'react-bootstrap';
import CONSTANTS from '../constants';
import Image from './Image';
import Recommendation from './Recommendation';
import './Movie.css';

const MAX_OVERVIEW_WORDS = 40;
const { SIZE } = CONSTANTS.IMAGE;

const getLabelInfo = (vote) => {
  let label = 'label-danger';

  if (vote > 5) {
    label = 'label-warning';
  }

  if (vote > 6) {
    label = 'label-info';
  }

  if (vote > 7) {
    label = 'label-success';
  }

  return label;
};

const Movie = (props) => {
  const {
    id,
    title,
    vote_average: voteAverage,
    overview,
    poster_path: posterPath,
    release_date: releaseDate,
  } = props.data;

  const overviewSmaller = overview
    .split(' ')
    .splice(0, MAX_OVERVIEW_WORDS)
    .join(' ');

  const releaseDateNormalized = new Date(releaseDate)
    .toLocaleString('en-us', {
      day: 'numeric',
      month: 'long',
      year: 'numeric',
    });

  const voteAverageNormalized = Math.floor(voteAverage);

  const labelInfo = getLabelInfo(voteAverageNormalized);

  return (
    <Col sm={12} lg={6}>
      <div className="Movie-card">
        <Row>
          <Col sm={12} md={4}>
            <Link to={`/details/${id}`}>
              <Image size={SIZE.SMALL} path={posterPath} title={title} />
            </Link>
          </Col>

          <Col sm={12} md={8}>
            <div className="Movie-body">

              <Media>
                <Media.Left>
                  <span className={`label ${labelInfo}`}>{voteAverage}</span>
                </Media.Left>

                <Media.Body>
                  <Media.Heading>
                    <Link to={`/details/${id}`}>
                      <span className="Movie-title">{title}</span>
                    </Link>
                    <br />
                    <small>
                      {releaseDateNormalized}
                    </small>
                  </Media.Heading>
                </Media.Body>
              </Media>

              <div className="Movie-overview">
                <p>{overviewSmaller}...</p>
              </div>

              <div className="Movie-details">
                <Link to={`/details/${id}`} className="Movie-details-link">
                  <span className="glyphicon glyphicon-info-sign" aria-hidden="true" />
                  &nbsp;
                  More info
                </Link>

                <Recommendation vote={voteAverageNormalized} />
              </div>
            </div>
          </Col>
        </Row>
      </div>
    </Col>
  );
};

Movie.propTypes = {
  data: PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    vote_average: PropTypes.number.isRequired,
    overview: PropTypes.string.isRequired,
    poster_path: PropTypes.string,
    release_date: PropTypes.string.isRequired,
  }).isRequired,
};

export default Movie;
