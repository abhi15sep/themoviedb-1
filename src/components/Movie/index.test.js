import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router';
import Movie from './index';

const movie = {
  id: 3,
  vote_average: 9,
  title: 'example 3',
  overview: 'Test',
  poster_path: '/path/to/image',
  release_date: '2017-04-28',
};

test('renders correctly', () => {
  const component = renderer.create(<Router><Movie key={movie.key} data={movie} /></Router>);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
