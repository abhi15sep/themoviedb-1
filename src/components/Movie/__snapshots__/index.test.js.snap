// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="col-lg-6 col-sm-12"
>
  <div
    className="Movie-card"
  >
    <div
      className="row"
    >
      <div
        className="col-md-4 col-sm-12"
      >
        <a
          href="/details/3"
          onClick={[Function]}
        >
          <img
            alt="example 3"
            src="http://image.tmdb.org/t/p/w185/path/to/image"
            title="example 3"
          />
        </a>
      </div>
      <div
        className="col-md-8 col-sm-12"
      >
        <div
          className="Movie-body"
        >
          <div
            className="row"
          >
            <div
              className="col-1"
            >
              <span
                className="badge badge-success"
              >
                9
              </span>
            </div>
            <div
              className="col-10"
            >
              <a
                href="/details/3"
                onClick={[Function]}
              >
                <span
                  className="Movie-title"
                >
                  example 3
                </span>
              </a>
              <br />
              <small>
                April 28, 2017
              </small>
            </div>
          </div>
          <p
            className="Movie-overview"
          >
            Test
            ...
          </p>
          <div
            className="Movie-details"
          >
            <a
              className="Movie-details-link"
              href="/details/3"
              onClick={[Function]}
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
                />
              </svg>
              Â 
              <small>
                More info
              </small>
            </a>
            <div
              className="Recommendation-main"
            >
              <small>
                <svg
                  className="Recommendation-icon"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M104 224H24c-13.255 0-24 10.745-24 24v240c0 13.255 10.745 24 24 24h80c13.255 0 24-10.745 24-24V248c0-13.255-10.745-24-24-24zM64 472c-13.255 0-24-10.745-24-24s10.745-24 24-24 24 10.745 24 24-10.745 24-24 24zM384 81.452c0 42.416-25.97 66.208-33.277 94.548h101.723c33.397 0 59.397 27.746 59.553 58.098.084 17.938-7.546 37.249-19.439 49.197l-.11.11c9.836 23.337 8.237 56.037-9.308 79.469 8.681 25.895-.069 57.704-16.382 74.757 4.298 17.598 2.244 32.575-6.148 44.632C440.202 511.587 389.616 512 346.839 512l-2.845-.001c-48.287-.017-87.806-17.598-119.56-31.725-15.957-7.099-36.821-15.887-52.651-16.178-6.54-.12-11.783-5.457-11.783-11.998v-213.77c0-3.2 1.282-6.271 3.558-8.521 39.614-39.144 56.648-80.587 89.117-113.111 14.804-14.832 20.188-37.236 25.393-58.902C282.515 39.293 291.817 0 312 0c24 0 72 8 72 81.452z"
                  />
                </svg>
                Editor choice
              </small>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
